modules/nlu/src/backend/engine.ts

```diff
@@ -25,13 +25,22 @@ export type TrainingOptions = {
   forceTrain: boolean
 }
 
+type UntrainedModel = Omit<Partial<Model>, 'data'> & {
```

Comment:
Sorry, I should have told you yesterday, but the more I look at this, the more it seems weird,

This would be less confusing:

```
type UntrainedModel = Omit<Model, 'data'> & {
  data: { input: TrainInput }
}

...

let model: Partial<UntrainedModel> = ....
```